solr/core/src/java/org/apache/solr/handler/designer/SchemaDesignerConfigSetHelper.java

```diff
@@ -128,43 +127,27 @@ class SchemaDesignerConfigSetHelper implements SchemaDesignerConstants {
   Map<String, Object> analyzeField(String configSet, String fieldName, String fieldText)
       throws IOException {
     final String mutableId = getMutableId(configSet);
-    final URI uri;
-    try {
-      uri =
-          collectionApiEndpoint(mutableId, "analysis", "field")
-              .setParameter(CommonParams.WT, CommonParams.JSON)
-              .setParameter("analysis.showmatch", "true")
-              .setParameter("analysis.fieldname", fieldName)
-              .setParameter("analysis.fieldvalue", "POST")
-              .build();
-    } catch (URISyntaxException e) {
-      throw new SolrException(SolrException.ErrorCode.BAD_REQUEST, e);
-    }
-
-    Map<String, Object> analysis = Collections.emptyMap();
-    HttpPost httpPost = new HttpPost(uri);
-    httpPost.setHeader("Content-Type", "text/plain");
-    httpPost.setEntity(new ByteArrayEntity(fieldText.getBytes(StandardCharsets.UTF_8)));
+    var solrParams = new ModifiableSolrParams();
+    solrParams.add("analysis.showmatch", "true");
+    solrParams.add("analysis.fieldname", fieldName);
+    solrParams.add("analysis.fieldvalue", "POST");
+    var request = new GenericSolrRequest(SolrRequest.METHOD.POST, "/analysis/field", solrParams);
+    request.withContent(fieldText.getBytes(StandardCharsets.UTF_8), "text/plain");
+    request.setRequiresCollection(true);
+    request.setResponseParser(new InputStreamResponseParser(null));
```

Comment:
```suggestion
    request.setResponseParser(new InputStreamResponseParser("json"));
```

Reply:
Actually, might `JsonMapResponseParser` better semantically represent what you want to do and maybe also cuts down on the lines of code needed?  How did you find InputStreamResponseParser anyway; that thing is very weird/awkward.